### DISCUSSION TITLE
Recreate original jpeg with new image

### DISCUSSION THREAD

**Discussion by brian-mcnamara** [ImageMagick/ImageMagick#4632] [Help]
Hey all,

Quite new to image processing so forgive me for what I assume is basic processing.

TLDR, I'm trying to recreate a images settings onto a new image and noticing that Resolution and Units are specified in my image while they are absent/undefined in the original. Can anyone help point me to what I'm missing?

Original image:
```
  Format: JPEG (Joint Photographic Experts Group JFIF format)
  Mime type: image/jpeg
  Class: DirectClass
  Geometry: 848x304+0+0
  Units: Undefined
  Colorspace: sRGB
  Type: TrueColor
  Base type: Undefined
  Endianess: Undefined
  Depth: 8-bit
```

My image:
```
  Format: JPEG (Joint Photographic Experts Group JFIF format)
  Mime type: image/jpeg
  Class: DirectClass
  Geometry: 848x304+0+0
  Resolution: 848x304 <------ This is not set above
  Print size: 1x1
  Units: PixelsPerInch <------ This is undefined above
  Colorspace: sRGB
  Type: TrueColor
  Base type: Undefined
  Endianess: Undefined
  Depth: 8-bit
```

Ive been using:
```
convert new.jpg -units undefined -density 848x304 -quality 80 -sampling-factor 2x2  new2.jpg
```
and have played around with some other options, but no matter what, I can't seem to strip the resolution or the units. Maybe it does not matter in the end.

Version: `Version: ImageMagick 6.9.10-23`

**Comment by fmw42** [comment]
Resolution and units are not relevant to a raster image for display. They are relevant when printing to define how large a pixel should print on paper. The input has no undefined resolution and no units because it was not set by the tool or camera. It is generally assumed to be 72 dpi. If you want to set it in your input or output use 

-density XX -units pixelsperinch

**Comment by snibgo** [comment]
>  ... I can't seem to strip the resolution or the units.

I dont think IM can do that. I expect exiftool can.

**Comment by brian-mcnamara** [comment]
Thanks both, I was under the impression based on the doc specifying `-units undefined` would remove the info, but it sounds like this is not the case. 

After some testing, it turned out the issue was not with the DPI or density, but the quality needed to come down to get it to render (guessing there is a max image size on the host). 

Anyway thanks for the info regarding how the attributes are used for JPEG.

**Reply by fmw42** [reply]
For reducing file size for JPG you can use  

`-quality x`

or

`-define jpeg:extent={size}  where size is in kb`


See 

https://legacy.imagemagick.org/Usage/formats/#jpg_write

and

https://imagemagick.org/script/command-line-options.php#quality
